### ISSUE TITLE
Traceback when trailing slash on workflow id

### ISSUE THREAD

**Issue by MetRonnie** [cylc/cylc-flow#4583]
Introduced by universal ID PR #3931 
```
$ cylc clean myflow/
Traceback (most recent call last):
  File "~/.conda/envs/cylc8/bin/cylc", line 33, in <module>
    sys.exit(load_entry_point('cylc-flow', 'console_scripts', 'cylc')())
  File "~/github/cylc-flow/cylc/flow/scripts/cylc.py", line 637, in main
    execute_cmd(command, *cmd_args)
  File "~/github/cylc-flow/cylc/flow/scripts/cylc.py", line 263, in execute_cmd
    COMMANDS[cmd].resolve()(*args)
  File "~/github/cylc-flow/cylc/flow/terminal.py", line 226, in wrapper
    wrapped_function(*wrapped_args, **wrapped_kwargs)
  File "~/github/cylc-flow/cylc/flow/scripts/clean.py", line 193, in main
    asyncio.run(run(*ids, opts=opts))
  File "~/.conda/envs/cylc8/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "~/.conda/envs/cylc8/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "~/github/cylc-flow/cylc/flow/scripts/clean.py", line 164, in run
    workflows, multi_mode = await parse_ids_async(
  File "~/github/cylc-flow/cylc/flow/id_cli.py", line 264, in parse_ids_async
    tokens_list.extend(_parse_cli(*ids))
  File "~/github/cylc-flow/cylc/flow/id_cli.py", line 125, in _parse_cli
    tokens = Tokens(id_)
  File "~/github/cylc-flow/cylc/flow/id.py", line 98, in __init__
    kwargs = tokenise(
  File "~/github/cylc-flow/cylc/flow/id.py", line 660, in tokenise
    raise ValueError(f'Invalid Cylc identifier: {identifier}')
ValueError: Invalid Cylc identifier: myflow/
```

Note that while tackling #4566 will help avoid getting a trailing slash in future, this is still a separate issue that needs to be addressed.

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

**Comment by oliver-sanders** [comment]
It's not a bug it's in the proposal!

**Comment by MetRonnie** [comment]
Well it is a bug that traceback is being shown 😏
